bindings/java/src/org/sleuthkit/datamodel/Image.java

```diff
@@ -377,6 +382,61 @@ public String getDeviceId() {
 	 */
 	@Override
 	public long getContentSize(SleuthkitCase sleuthkitCase) {
-		return AbstractDataSource.getContentSize(sleuthkitCase, getId());
+		SleuthkitCase.CaseDbConnection connection;
+		Statement statement = null;
+		ResultSet resultSet = null;
+		long contentSize = 0;
+
+		try {
+			connection = sleuthkitCase.getConnection();
+
+			try {
+				statement = connection.createStatement();
+				resultSet = connection.executeQuery(statement, "SELECT SUM (size) FROM tsk_image_info WHERE tsk_image_info.obj_id = " + getId());
+				if (resultSet.next()) {
+					contentSize = resultSet.getLong("sum");
+				}
+			} catch (SQLException ex) {
+				LOGGER.log(Level.SEVERE, String.format("There was a problem while querying the database for size data for object ID %d.", getId()), ex); //NON-NLS
+			} finally {
+				closeResultSet(resultSet);
+				closeStatement(statement);
+				connection.close();
+			}
+		} catch (TskCoreException ex) {
+			LOGGER.log(Level.SEVERE, "An error occurred attempting to establish a database connection.", ex); //NON-NLS
+		}
+
+		return contentSize;
+	}
```

Comment:
Looking at this method causes me to realize I made a mistake when specifying the getContentSize API. It should throw TskCoreException, including wrapping any SQLException in a TskCoreExcepton.  Sorry about the oversight, please update the code throughout as required, thanks!